vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -1776,6 +1794,7 @@ func (vm *virtualMachine) setupVirtualMachine(c *govmomi.Client) error {
 		Flags: &types.VirtualMachineFlagInfo{
 			DiskUuidEnabled: &vm.enableDiskUUID,
 		},
+		Annotation: vm.annotation,
```

Comment:
No, this doesn`t fail with empty string in vm.annotaton. So everything works fine as with an empty argument and absolutely without it. In both of these situations vm.annotation takes a value "". [vsphere/resource_vsphere_virtual_machine.go (L710-L714)](https://github.com/verdel/terraform-provider-vsphere/blob/8624dabc7e5517fbd09eeaf78b25c7eb74f6193f/vsphere/resource_vsphere_virtual_machine.go#L710-L714)